app/views/shared/_post.html.erb

```diff
@@ -11,4 +11,9 @@
     <p>Likes: <%= post.likes_counter || 0 %></p>
   </div>
 </div>
+
+<% if can? :delete, @post %>
+    <button type="button" class="delete_btn btn"><%= link_to "Delete Post", user_post_path(user_id: @post.author.id, id: @post.id), data: { turbo_method: :delete } %></button>
+  <% end %>
+  
```

Comment:
- [ Required ] Great job, implementing the `delete post` functionality. However, when I log in as an `admin` and try to access a user's posts I face the error below. Please see the screenshot below to understand what I'm talking about and rectify this.

## Error when I visit http://localhost:3000/users/{user_id}/posts
![Screenshot from 2023-02-11 17-03-42](https://user-images.githubusercontent.com/16392046/218262253-782aa84b-73e4-4eb1-bd1e-01e978b47139.png)

I think you should change the code as shown below, for the page to work as expected.
Instead of `@post` use `post`
```ruby
<% if can? :delete, post %>
    <button type="button" class="delete_btn btn">
    <%= link_to "Delete Post", user_post_path(user_id: post.author.id, id: post.id), data: { turbo_method: :delete } %>
    </button>
  <% end %>
```

Reply:
Thank you for the kind suggestion